adapters/mqtt-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -1303,6 +1304,22 @@ private Future<Void> checkTopic(final MqttContext context) {
             }
         }
 
+        private boolean disconnectOnExpired() {
+            if (authenticatedDevice != null && authenticatedDevice.expired()) {
```

Comment:
IMHO we should move this condition to e.g. *boolean MqttContext.isExpired()* ...

Reply:
My bad, I was under the (wrong) impression that we always have an MqttContext for all types of MQTT packets we process ...